### ISSUE TITLE
net/http: tls with large number of connections

### ISSUE THREAD

**Issue by stomir** [golang/go#16878]
Please answer these questions before submitting your issue. Thanks!

1. What version of Go are you using (`go version`)?
go version go1.7 linux/amd64

2. What operating system and processor architecture are you using (`go env`)?
go version go1.7 linux/amd64

3. What did you do?
Start a HTTPS server with ListenAndServeTLS

4. What did you expect to see?
Handling a lot of short connections

5. What did you see instead?
Not handling a lot of short connections

So basically, the TLS handler is bad at handling large number of short connections,
and works a lot better when I removed the "tcpKeepAliveListener" from this line:
tlsListener := tls.NewListener(tcpKeepAliveListener{ln.(*net.TCPListener)}, config)

(I disabled KeepAlives with SetKeepAlivesEnabled anyway)

**Comment by quentinmit** [comment]
I'm afraid we need more information. Can you write a simple benchmark that demonstrates the problem and show how much faster it gets with the change you propose?

**Comment by stomir** [comment]
ok, this is pretty embarrassing

When I was working on the problem, at some point I was forced to update go version on the machine I was working on, and that was what actually helped me.
So this was probably some already resolved problem with handling cryptography in only one thread.
Sorry for wasting your time.
